### ISSUE TITLE
post-autoload-dump event returned with error code 1

### ISSUE THREAD

**Issue by aisakk** [clarkeash/doorman#57]
Hi guys, i have this problem when i execute composer require clarkeash/doorman and when use php artisan clear :/, how fix this problem? my version of laravel is 7.13.0

```
composer require clarkeash/doorman       
Using version ^5.0 for clarkeash/doorman
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 1 install, 0 updates, 0 removals
  - Installing clarkeash/doorman (v5.0.1): Downloading (100%)
Writing lock file
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
> @php artisan package:discover --ansi

   Error

  Class name must be a valid object or a string

  at C:\xampp\htdocs\giknomina\vendor\clarkeash\doorman\src\Providers\DoormanServiceProvider.php:46
    42|             'doorman'
    43|         );
    44|
    45|         $this->app->bind(BaseInvite::class, function($app) {
  > 46|             return new $app['config']['doorman.invite_model'];
    47|         });
    48|
    49|         $this->app->bind('doorman', Doorman::class);
    50|         $this->app->singleton(Doorman::class, Doorman::class);

  1   C:\xampp\htdocs\giknomina\vendor\laravel\framework\src\Illuminate\Container\Container.php:801
      Clarkeash\Doorman\Providers\DoormanServiceProvider::Clarkeash\Doorman\Providers\{closure}(Object(Illuminate\Foundation\Application), [])

  2   C:\xampp\htdocs\giknomina\vendor\laravel\framework\src\Illuminate\Container\Container.php:687
      Illuminate\Container\Container::build(Object(Closure))
Script @php artisan package:discover --ansi handling the post-autoload-dump event returned with error code 1

Installation failed, reverting ./composer.json to its original content.


Composer json
```js

 "require": {
        "php": "^7.2.5",
        "fideloper/proxy": "^4.2",
        "fruitcake/laravel-cors": "^2.0",
        "guzzlehttp/guzzle": "^6.3",
        "laravel/framework": "^7.0",
        "laravel/socialite": "^4.3",
        "laravel/tinker": "^2.0",
        "laravel/ui": "^2.0",
        "ramsey/uuid": "^3 || ^4"
    },
    "require-dev": {
        "facade/ignition": "^2.0",
        "fzaninotto/faker": "^1.9.1",
        "mockery/mockery": "^1.3.1",
        "nunomaduro/collision": "^4.1",
        "phpunit/phpunit": "^8.5"
    },
    "config": {
        "optimize-autoloader": true,
        "preferred-install": "dist",
        "sort-packages": true
    },
    "extra": {
        "laravel": {
            "dont-discover": []
        }
    },
    "autoload": {
        "psr-4": {
            "App\\": "app/"
        },
        "classmap": [
            "database/seeds",
            "database/factories"
        ]
    },
    "autoload-dev": {
        "psr-4": {
            "Tests\\": "tests/"
        }
    },
    "minimum-stability": "dev",
    "prefer-stable": true,
    "scripts": {
        "post-autoload-dump": [
            "Illuminate\\Foundation\\ComposerScripts::postAutoloadDump",
            "@php artisan package:discover --ansi"
        ],
        "post-root-package-install": [
            "@php -r \"file_exists('.env') || copy('.env.example', '.env');\""
        ],
        "post-create-project-cmd": [
            "@php artisan key:generate --ansi"
        ]
    }
```
```

**Comment by clarkeash** [comment]
Composer:

```bash
composer require clarkeash/doorman
Using version ^5.0 for clarkeash/doorman
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 1 install, 0 updates, 0 removals
  - Installing clarkeash/doorman (v5.0.1): Downloading (100%)
Writing lock file
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
> @php artisan package:discover --ansi
Discovered Package: clarkeash/doorman
Discovered Package: facade/ignition
Discovered Package: fideloper/proxy
Discovered Package: fruitcake/laravel-cors
Discovered Package: laravel/tinker
Discovered Package: nesbot/carbon
Discovered Package: nunomaduro/collision
Package manifest generated successfully.
```

Laravel:

```bash
php artisan --version
Laravel Framework 7.14.1
```

PHP:

```bash
php -v
PHP 7.4.4 (cli) (built: Mar 19 2020 20:12:27) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.4, Copyright (c), by Zend Technologies
```

So I have just installed it and it seems to work.

**Comment by clarkeash** [comment]
@Aisakk Ahh, you're on windows?

This library has not been built with windows in mind, thats something I can fix.

**Comment by aisakk** [comment]
clarkeash, yes i use windows :/, thanks to answer.

**Comment by clarkeash** [comment]
It should be a fairly simple fix for me, but it will have to wait until the weekend.

**Comment by aisakk** [comment]
ok, alright, thanks for the answer.

**Comment by clarkeash** [comment]
Okay I guess you dont want it fixed

**Comment by aisakk** [comment]
Mm, yeah, I want you fixed the package, if you don't have problem, about that.

**Comment by clarkeash** [comment]
Okay.

**Comment by clarkeash** [comment]
@Aisakk I have ran this on windows and everything seems to be working fine.
Do you have an existing config file in `config/doorman.php` ?

**Comment by aisakk** [comment]
Hi clark, no, i dont. Need i to create this file, then download the package? or upside down?

**Comment by aisakk** [comment]
Im install the package, and create the file doorman.php in the directory config, i had the same result,the content in this file i have:
```
return [
    'invite_table_name' => 'invites',
]; 
```
And using alias and provider in the file config/app.php , but i had the same result at error  in this isseous. I´m install the package, then comes out this: 


```
   Error

  Class name must be a valid object or a string

  at C:\xampp\htdocs\giknomina\vendor\clarkeash\doorman\src\Providers\DoormanServiceProvider.php:46
    42|             'doorman'
    43|         );
    44|
    45|         $this->app->bind(BaseInvite::class, function($app) {
  > 46|             return new $app['config']['doorman.invite_model'];
    47|         });
    48|
    49|         $this->app->bind('doorman', Doorman::class);
    50|         $this->app->singleton(Doorman::class, Doorman::class);

  1   C:\xampp\htdocs\giknomina\vendor\laravel\framework\src\Illuminate\Container\Container.php:801
      Clarkeash\Doorman\Providers\DoormanServiceProvider::Clarkeash\Doorman\Providers\{closure}(Object(Illuminate\Foundation\Application), [])

  2   C:\xampp\htdocs\giknomina\vendor\laravel\framework\src\Illuminate\Container\Container.php:687
      Illuminate\Container\Container::build(Object(Closure))
```

**Comment by clarkeash** [comment]
Your `doorman.php` file should look like https://raw.githubusercontent.com/clarkeash/doorman/master/resources/config/doorman.php

Most importantly you should have an invite model defined like so:

```php
'invite_model' => Clarkeash\Doorman\Models\Invite::class,
```

**Comment by aisakk** [comment]
Hi bro, i copy your code of the link you passed me, but I keep getting the same error. 
![image](https://user-images.githubusercontent.com/43797095/83949678-3ec1c800-a7f3-11ea-9b9c-9d72c7dad859.png)

**Comment by aisakk** [comment]
![image](https://user-images.githubusercontent.com/43797095/83949694-539e5b80-a7f3-11ea-8b95-a819961dcefa.png)

**Comment by Mahmoudjs** [comment]
i have the same issue in laravel 6.2, did you fix it ?

**Comment by aisakk** [comment]
No bro, im wait to answer of my partner clark.

**Comment by clarkeash** [comment]
@Mahmoudjs Are you also on windows?

**Comment by Mahmoudjs** [comment]
Yes

**Comment by Mahmoudjs** [comment]
I have tried to add log in this code 
```
 $this->app->bind(BaseInvite::class, function($app) {
            Log::info($app['config']['doorman.invite_model']);
            return new $app['config']['doorman.invite_model'];
        });
```
but Log::info($app['config']['doorman.invite_model']); return null in log file

**Comment by elsayed-mazen** [comment]
Have the same problem in linux .. which is not recognizing $app['config']['doorman.invite_model']
![doorman installation error](https://user-images.githubusercontent.com/666768/83973570-0bb12e80-a8f0-11ea-9637-a203c89c93b1.jpg)

**Comment by clarkeash** [comment]
I am confused by this, what version of PHP is everyone using?

**Comment by Mahmoudjs** [comment]
php -v 7.2 
I tried to copy the doorman.php config file and add it to my project before installing the packege and run the command php artisan config:cache , then installing the package after that , and everything worked fine.

**Comment by aisakk** [comment]
I have php 7.4 in xampp

**Comment by clarkeash** [comment]
This may have been fixed in #64 if not there is not much more I can do, I dont dev on windows and I was unable to reproduce it with github actions running on windows.

Feel free to open a new issue if it still persists, but I will need a PR for a fix as I dont know what to do.

**Comment by debrawais** [comment]
Script @php artisan package:discover --ansi handling the post-autoload-dump event returned with error code 255       

Installation failed, reverting ./composer.json and ./composer.lock to their original content.


Tengo este error al momento de querrer instalar Livewire, o poner el composer update.
